drivers/comparator/comparator_stm32_comp.c

```diff
@@ -0,0 +1,324 @@
+{
+#ifdef CONFIG_PM_DEVICE
+	enum pm_device_state state;
+
+	(void)pm_device_state_get(DEVICE_DT_INST_GET(0), &state);
+	return state == PM_DEVICE_STATE_ACTIVE;
+#else
+	return true;
+#endif /* CONFIG_PM_DEVICE */
+}
+
+static int stm32_comp_get_output(const struct device *dev)
+{
+	const struct stm32_comp_config *cfg = dev->config;
+	COMP_TypeDef *comp = cfg->comp;
+
+	return LL_COMP_ReadOutputLevel(comp);
+}
+
+static int stm32_comp_set_trigger(const struct device *dev, enum comparator_trigger trigger)
+{
+	const struct stm32_comp_config *cfg = dev->config;
+	struct stm32_comp_data *data = dev->data;
+	stm32_exti_trigger_type exti_trigger = 0U;
+	COMP_TypeDef *comp = cfg->comp;
+	int ret = 0;
+
+	switch (trigger) {
+	case COMPARATOR_TRIGGER_NONE:
+		exti_trigger = STM32_EXTI_TRIG_NONE;
+		break;
+	case COMPARATOR_TRIGGER_RISING_EDGE:
+		exti_trigger = STM32_EXTI_TRIG_RISING;
+		break;
+	case COMPARATOR_TRIGGER_FALLING_EDGE:
+		exti_trigger = STM32_EXTI_TRIG_FALLING;
+		break;
+	case COMPARATOR_TRIGGER_BOTH_EDGES:
+		exti_trigger = STM32_EXTI_TRIG_BOTH;
+		break;
+	default:
+		LOG_DBG("%s: Unsupported trigger mode %d", dev->name, trigger);
+		return -ENOTSUP;
+	}
+
+	irq_disable(cfg->irq_nr);
+	LL_COMP_Disable(comp);
+
+	ret = stm32_exti_enable(cfg->exti_line_number, exti_trigger,
+				STM32_EXTI_MODE_IT);
+	if (ret != 0) {
+		LOG_DBG("%s: EXTI init failed (%d)", dev->name, ret);
+		return ret;
+	}
+
+	if (stm32_comp_is_resumed()) {
+		LL_COMP_Enable(comp);
+	}
+
+	if (data->callback != NULL) {
+		irq_enable(cfg->irq_nr);
+	}
+
+	return ret;
+}
+
+static int stm32_comp_trigger_is_pending(const struct device *dev)
+{
+	const struct stm32_comp_config *cfg = dev->config;
+
+	if (stm32_exti_is_pending(cfg->exti_line_number)) {
+		stm32_exti_clear_pending(cfg->exti_line_number);
+		return 1;
+	}
+	return 0;
+}
+
+static int stm32_comp_set_trigger_callback(const struct device *dev, comparator_callback_t callback,
+					   void *user_data)
+{
+	const struct stm32_comp_config *cfg = dev->config;
+	struct stm32_comp_data *data = dev->data;
+
+	irq_disable(cfg->irq_nr);
+
+	data->callback = callback;
+	data->user_data = user_data;
+
+	irq_enable(cfg->irq_nr);
+
+	if (data->callback != NULL && stm32_comp_trigger_is_pending(dev)) {
+		callback(dev, user_data);
+	}
+
+	return 0;
+}
+
+static DEVICE_API(comparator, stm32_comp_comp_api) = {
+	.get_output = stm32_comp_get_output,
+	.set_trigger = stm32_comp_set_trigger,
+	.set_trigger_callback = stm32_comp_set_trigger_callback,
+	.trigger_is_pending = stm32_comp_trigger_is_pending,
+};
+
+static int stm32_comp_pm_callback(const struct device *dev, enum pm_device_action action)
+{
+	const struct stm32_comp_config *cfg = dev->config;
+	COMP_TypeDef *comp = cfg->comp;
+
+	if (action == PM_DEVICE_ACTION_RESUME) {
+		if (cfg->lock_enable) {
+			LL_COMP_Lock(comp);
+		}
+		LL_COMP_Enable(comp);
+	}
+
+#ifdef CONFIG_PM_DEVICE
+	if (action == PM_DEVICE_ACTION_SUSPEND) {
+		LL_COMP_Disable(comp);
+	}
+#endif
```

Comment:
Suggestion as it's preferable to use `if (IS_EANBLED())` instead of `#ifdef` when possible:$
```suggestion
	if (IS_ENABLED(CONFIG_PM_DEVICE) && action == PM_DEVICE_ACTION_SUSPEND) {
		LL_COMP_Disable(comp);
	}